src/Mail/BaseMail.php

```diff
@@ -0,0 +1,37 @@
+<?php
+
+namespace RonasIT\Support\Mail;
+
+use Illuminate\Bus\Queueable;
+use Illuminate\Contracts\Queue\ShouldQueue;
+use Illuminate\Mail\Mailable;
+use Illuminate\Queue\SerializesModels;
+
+class BaseMail extends Mailable implements ShouldQueue
+{
+    use Queueable;
+    use SerializesModels;
+
+    public int $tries = 5;
+    public string $queue = 'mails';
+
+    public function __construct(array $viewData, $view, $subject)
+    {
+        $this->viewData = $viewData;
+        $this->subject = $subject;
+        $this->view = $view;
+        $this->onQueue($this->queue);
```

Comment:
```suggestion
        $this->onQueue($this->queue ?? 'mails');
```

Reply:
I made it simplier. I decide we simply could override the queue if it necessary.

```php
class BaseMail extends Mailable implements ShouldQueue
{
    use Queueable;
    use SerializesModels;

    public int $tries = 5;

    public function __construct(array $viewData, $subject, $view)
    {
        $this->viewData = $viewData;
        $this->subject = $subject;
        $this->view = $view;
        $this->queue = 'mails';
    }
}
```

```php
class SomeMail extends BaseMail
{
    public function __construct(array $viewData)
    {
       parent::__construct(
            $viewData,
            'Subject of this mail',
            'emails.my_new_template'
        );

        $this->queue = 'different_queue';
    }
}
```